plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/BaseProjectImpl.java

```diff
@@ -186,11 +188,17 @@ public boolean isPrivateProject() {
     @NotNull
     @Override
     public DBPDataSourceRegistry getDataSourceRegistry() {
-        ensureOpen();
-        synchronized (metadataSync) {
-            if (dataSourceRegistry == null) {
-                dataSourceRegistry = createDataSourceRegistry();
-            }
+        if (dataSourceRegistry == null) {
+            RuntimeUtils.runTask(monitor -> {
+                if (dataSourceRegistry == null) {
+                    synchronized (metadataSync) {
+                        ensureOpen();
+                        if (dataSourceRegistry == null) {
```

Comment:
The multiple calls of the same condition` "if (dataSourceRegistry == null) {...}"` looks a little bit strange.